baseplate/experiments/providers/variant_sets/multi_variant_set.py

```diff
@@ -0,0 +1,66 @@
+from .base import VariantSet
+
+
+class MultiVariantSet(VariantSet):
+    """ Variant Set designed to handle more than two total treatments.
+
+    MultiVariantSets are not designed to support changes in variant sizes without
+    rebucketing.
+    """
+
+    def __init__(self, variants, num_buckets=1000):
+        """ :param list variants: Array of dicts, each containing the keys 'name'
+            and 'size'. Name is the variant name, and size is the fraction of
+            users to bucket into the corresponding variant. Sizes are expressed
+            as a floating point value between 0 and 1.
+        :param int num_buckets: The number of potential buckets that can be
+            passed in for a variant call. Defaults to 1000, which means maximum
+            granularity of 0.1% for bucketing
+        """
+        self.num_buckets = num_buckets
+        self.variants = variants
+
+        self._validate_variants()
+
+    def __contains__(self, item):
+        for variant in self.variants:
+            if variant.get('name') == item:
+                return True
+
+        return False
+
+    def _validate_variants(self):
+
+        if self.variants is None:
+            raise ValueError('No variants provided')
+
+        if len(self.variants) < 3:
+            raise ValueError("MultiVariant experiments expect three or "
+                "more variants.")
+
+        total_size = 0.0
```

Comment:
I think you want to make this an `total_size = 0` so it doesn't get turned into a float.